File: 2785.sort-vowels-in-a-string/solution.go
func sortVowels(s string) string {
    var vowels []byte
    
    for i := range s {
        if isVowel(s[i]) {
            vowels = append(vowels, s[i])
        }
    }
    sort.Slice(vowels, func(i, j int) bool { return vowels[i] < vowels[j] })
    
    res := []byte(s)
    k := 0 
    for i := range s {
        if isVowel(s[i]) {
            res[i] = vowels[k]
            k++
        }
    }
    return string(res)
}

func isVowel(c byte) bool {
    return strings.IndexByte("aeiouAEIOU", c) != -1
}

File: 2785.sort-vowels-in-a-string/solution.kt
class Solution {
    fun sortVowels(s: String): String {
        val vowels = s.asIterable().filter { it.isVowel() }.sorted()
        var vowelsIterator = vowels.iterator()
        return s.map { ch ->
            if (ch.isVowel()) {
                vowelsIterator.next()
            } else {
                ch
            }
        }.joinToString("")
    }
}

fun Char.isVowel(): Boolean = "AEIOUaeiou".indexOf(this) != -1